cmd/lifecycle/analyzer.go

```diff
@@ -172,3 +216,11 @@ func (a *analyzeCmd) registryImages() []string {
 	}
 	return registryImages
 }
+
+func (a *analyzeCmd) restoresLayerMetadata() bool {
+	return api.MustParse(a.platform.API()).Compare(api.MustParse("0.7")) < 0
+}
+
+func (a *analyzeCmd) platformAPIVersionGreaterThan06() bool {
+	return api.MustParse(a.platform.API()).Compare(api.MustParse("0.7")) >= 0
+}
```

Comment:
Please see [this comment](https://github.com/buildpacks/lifecycle/pull/582#discussion_r615973922).

Reply:
Like you said, these were created at different times and different purposes. I thought it might read better to see code like ` if restoresLayerMetadata { doRestore() }` then `if PlatformVersionGreaterThan07 { doRestore() }`, but I ran afoul of that when I ended up needing to guard things that were a shuffling of inputs rather than a feature guard.

I wonder if we should introduce an easier way to guard things by version. Maybe make `platform` or `apiVersion` comparable so we can write more terse `if a.Platform > platform.0_7 { doRestore() }`

What do you think?

Reply:
In general I think we should try to be as descriptive as possible. Made a suggestion here: https://github.com/buildpacks/lifecycle/pull/582/files#r616685220

But for `DefineFlags` probably switching off of version is fine.

Reply:
Uggh my link did not work. I'd actually be in favor of adding an additional function `a.supportsImageArgument()` to use when parsing the image. We would be essentially duplicating `api.MustParse(a.platform.API()).Compare(api.MustParse("0.7"))` three times BUT the code would be a lot more readable IMO and I think that's a fair trade-off.

Reply:
Eventually we might consider having the `platform` interface provide this logic, since it is essentially a platform concern.

Reply:
@natalieparellano, I agree that we should be as descriptive as possible.
The only thing that bothers me in this case is having 2 functions that are doing almost the same.
My suggestion is to choose a good function name (what Natalie suggested sounds good to me) and use it in all cases, instead of having these 2 functions.
What do you think?

Reply:
IDK, if you look at the other place where this is used: https://github.com/buildpacks/lifecycle/blob/f92acf0fd513a6b059f5eae41f19d660fc37635e/cmd/lifecycle/analyzer.go#L61

`!a.supportsImageArgument()` would be weird there.

Reply:
mmmm...yeah, maybe.
I mostly didn't like what happened in `DefineFlags` where we have:
```
if a.restoresLayerMetadata() {
...
}
...
if a.platformAPIVersionGreaterThan06() {
...
}
```
I think it's very confusing.

What do you think about having only the function `platformAPIVersionGreaterThan06`, and add a comment at the top with a description of what is supported starting at api 0.7.

I'm not sure what is the best solution. I don't like a lot any of these ideas :\
I'll try to think about something else, but in the meantime, I would love to get your opinions :)

Reply:
I changed this in 71c8e55f1de14d28526c8db9789a152529b7e588, let me know what you think. I removed the feature guard from the `DefineFlags` but used feature guards in the code executing on those flags. I also chained the functions to make it clearer.

Reply:
Looks great!

Reply:
I agree! I like your change!